packages/vaex-core/vaex/column.py

```diff
@@ -68,6 +68,38 @@ def trim(self, i1, i2):
         return ColumnVirtualRange(self.start + i1 * self.step, self.start + i2 * self.step, self.step, self.dtype)
 
 
+class ColumnVirtualConstant(Column):
+    def __init__(self, value, length, dtype=None, chunk_size=1024):
+        self.value = value
+        self.length = length
+        self.dtype = self._get_dtype(dtype)
+        self.chunk_size = chunk_size
+
+    def __len__(self):
+        return self.length
+
+    def __getitem__(self, slice):
+        template = pa.array([self.value] * self.chunk_size, type=self.dtype)
+        n_chunks = (self.length + self.chunk_size - 1) // self.chunk_size
+        ar = pa.chunked_array([template] * n_chunks)
+        ar = ar.slice(length=self.length)
```

Comment:
this cannot be self.length, it should be related to `slice` (example are probably in the same .py file)

Reply:
i adjusted things, hope it is better now.